validations/src/validations.rs

```diff
@@ -2091,20 +2100,31 @@ pub fn calculate_reppoe_threshold(
     rep_eng: &ReputationEngine,
     pkh: &PublicKeyHash,
     num_witnesses: u16,
+    block_epoch: u32,
+    minimum_difficulty: u32,
 ) -> (Hash, f64) {
-    let total_active_rep = rep_eng.total_active_reputation();
-
     // Add 1 to reputation because otherwise a node with 0 reputation would
     // never be eligible for a data request
-    let my_reputation = u64::from(rep_eng.get_eligibility(pkh)) + 1;
+    let my_eligibility = u64::from(rep_eng.get_eligibility(pkh)) + 1;
     let factor = u64::from(rep_eng.threshold_factor(num_witnesses));
 
     let max = u64::max_value();
-    // Check for overflow: when the probability is more than 100%, cap it to 100%
-    let target = if my_reputation.saturating_mul(factor) >= total_active_rep {
-        max
+    // Compute target eligibility and hard-cap it if required
+    let target = if after_third_hard_fork(block_epoch, get_environment()) {
+        // TODO: Review if next line is required (check if total active reputation could be zero)
+        let total_active_rep = std::cmp::max(rep_eng.total_active_reputation(), 1);
+        // If eligibility is more than 100%, cap it to (max/minimum_difficulty*factor)%
```

Comment:
I think this comment is wrong?

```suggestion
        // Eligibility must never be greater than (max/minimum_difficulty)
```

Reply:
Do you want to commit this suggestion?